File: test/data/cs/test01/StateState1.cs
/// <summary>
/// this is a string
/// this is another string
/// </summary>
public class StateState1 : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateState1(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitCommand1()
    {
        _controllee.DoAction1();
        return null;
    }
    public override BaseState? TransitCommand2()
    {
        _controllee.DoAction2();
        _stateController.InstanceOfState2.Setup(false, false);
        return _stateController.InstanceOfState2;
    }
    public override string GetStateName()
    {
        return "State1";
    }
    public override BaseState? GetParent()
    {
        return null;
    }
}

File: test/data/typescript/test01/StateState1.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
/// <summary>
/// this is a string
/// this is another string
/// </summary>
export class StateState1 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitCommand1(): BaseState | null
    {
        this._controlleeImp.DoAction1();
        return null;
    }
    public TransitCommand2(): BaseState | null
    {
        this._controlleeImp.DoAction2();
        this._stateController.InstanceOfState2.Setup();
        return this._stateController.InstanceOfState2;
    }
    public  GetStateName(): string
    {
        return "State1";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
}